### PR TITLE
Some "how to document" content

### PR DESCRIPTION
**Pull request by marians**
- Renamed `index.md` to `README.md`, since that one gets rendered automatically when browsing directories.
- Added some guidelines content. Will likely extend this once our products sub pages are done.

### CODE REVIEW

**Comment by marians** [comment]
RFR @puja108

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# How to document a Giant Swarm product
+
+We provide product-specific documentation inside the repository of the product, using a top-level directory named `docs`.
+
+The content within this directory is intended primarily for reading on the `giantswarm.io` website. To enable useful rendering in that context, please adhere to the following guidelines.
```
Comment (puja108) [inline docs/README.md]: I would remove `primarily` here as reading it on github will happen and is also wanted

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+## Markdown content
+
+Documentation content is in [Markdown](https://daringfireball.net/projects/markdown/syntax) format. Github-flavored Markdown extensions such as fenced code blocks and tables might be applied.
+
+## Index files
+
+We make use of the fact that when entering a directory of a repository, GitHub automatically renders the `README.md` file content if available.
+
+If no `README.md` file is available in the `docs` folder or any subfolder, a table of contents page is automatically generated for browsing that folder within the `giantswarm.io` website. To provide a custom table of contents, provide a `README.md` file and set links to all sub-pages.
```
Comment (puja108) [inline docs/README.md]: best recommend to always have a README.md so we have a better/coherent UX on GitHub

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+When rendering fenced code blocks, syntax highlighting is applied automatically. Some languages can be detected rather well. However, some are rather problematic and need help.
+
+Especially shell commands and their output regularly lead to bad syntax highlighting results. For that sort of code, syntax highlighting should therefore be disabled completely. There is a helper `nohighlight` that can be added to the according code block. Example:
+
+```nohighlight
+```nohighlight
+$ some shell code
+...
+```
```
Comment (puja108) [inline docs/README.md:57]: add another row of backticks?

**Comment by puja108** [comment]
other than my few inline comments LGTM

also ping @giantswarm/dev for acknowledgment in our current and future open source repos

**Comment by marians** [comment]
Thanks for reviewing, @puja108. A general comment:

I think we should strive for optimizing the docs experience on our site, while at the same time allowing for a fair experience on GitHub. Our means are much more limited on GitHub, and the effort required will be much higher, compared to what we can accomplish on our site. So I see GitHub mainly as the content repository and our site mainly as the reading interface.

Hence the "primarily" was chosen with intent.

File: docs/README.md (hunk 1 above)
Comment (marians) [inline docs/README.md]: > I would remove primarily here as reading it on github will happen and is also wanted

See my general comment

File: docs/README.md (hunk 2 above)
Comment (marians) [inline docs/README.md]: > best recommend to always have a README.md so we have a better/coherent UX on GitHub

I agree

File: docs/README.md (hunk 3 above)
Comment (marians) [inline docs/README.md:57]: I tried something like 6 different combinations of backticks and backslashes (for escaping) but couldn't make it work. Also with no backslashes at all. This was the only solution that worked. If you know how to find a way to escape the ending triple backticks, please let me know.

File: docs/README.md (hunk 1 above)
Comment (puja108) [inline docs/README.md]: Yes, I generally agree, just wanted it to be clear that there will be people looking at it through github and it should at least look decent in there.

File: docs/README.md (hunk 3 above)
Comment (puja108) [inline docs/README.md:57]: ok, thanks for trying that out anyway :)

**Comment by puja108** [comment]
Like mentioned in the comment, I do agree with you, I just feared it would maybe make people think that's the only place, but I guess with the rest of the text it should be clear.

**Comment by marians** [comment]
I clarified in the according places regarding primary/secondary and recommending `README.md`

**Pull request merged**